### DISCUSSION TITLE
Compare a workflow.out with a value

### DISCUSSION THREAD

**Discussion by Grelot** [nextflow-io/nextflow#2937] [Q&A] [answered]
Hi everyone,

First, thank a lot for the development of Nextflow, this is a great software.

I want to test if an output from a workflow have an excepted value to run another conditionnal workflow.

main.nf
```
include {CHECKING ; ANOTHER_WORKFLOW} from './workflows/check'

workflow {
    CHECKING()
    if(CHECKING.out == "valide") { ANOTHER_WORKFLOW() }

}
```

check.nf
```
include { CHECKLABFILE } from '../modules/local/labfile_check'

workflow CHECKING {
   main:
       CHECKLABFILE()
        if(CHECKLABFILE.out != "faulty_format") {
            checkingResult = "valide."
        } else {
            checkingResult = "faulty."
        }
    emit:
        checkingResult
}

```

../modules/local/labfile_check'
```
process CHECKLABFILE() {
    output:
        stdout
    script:
    """
    check.py $params.labfile
    """
}
```

The process is working but I don't manage to be able to compare the output value emitted by a workflow. Any advices ?

**Comment by bentsherman** [comment, marked as answer]
This is a common stumbling block when learning to code in a "reactive" or "functional" way. Basically, you should never try to "check" the value of a channel, instead you should use channel operators to process channel outputs as they are emitted.

In this case, I would use the [branch](https://www.nextflow.io/docs/latest/operator.html#branch) operator to pipe outputs from `CHECKLABFILE` into a "valid" channel or "faulty" channel depending on the output value. Then you can pipe the valid channel into the next workflow and the faulty channel into the void (or whatever).

**Reply by Grelot** [reply]
Thanks a lot, indeed I need to learn this new logic. I changed my whole workflow to remove "checking" of channel value inside process and write alternative path of processes instead.
